src/main/java/com/diffmin/App.java

```diff
@@ -209,6 +209,14 @@ private static void applyInsertion(ImmutableTriple<Integer, CtElement, CtElement
                 ((CtExecutable<?>) inWhichElement)
                         .addParameterAt(where, (CtParameter<?>) toBeInserted);
                 break;
+            case THROWN:
+                if (!(((CtExecutable<?>) inWhichElement).getThrownTypes()
+                        == ((CtExecutable<?>) toBeInserted.getParent()).getThrownTypes())) {
+                    ((CtExecutable<?>) inWhichElement)
+                            .setThrownTypes(
+                                    ((CtExecutable<?>) toBeInserted.getParent()).getThrownTypes());
+                }
```

Comment:
This solution relies on the implementation detail that `CtExecutableImpl` directly returns its internal set from the `getThrownTypes` method, i.e. it looks like this:

```java
... getThrownTypes() {
    return thrownTypes;
}
```

Whereas in other parts of Spoon, it looks something like this:

```java
... getThrownTypes() {
    return Collections.unmodifiableSet(thrownTypes);
}
```
If `getThrownTypes` were to be changed to return a copy or an unmodifiable set, the solution you propose would fail, as the object reference would be different. Instead of doing that, I suggest you either filter out duplicate operations before applying them (optimal but more effort), or just set a copy (suboptimal but easy). Here's what I mean by setting a copy:

```suggestion
                var thrownTypesCopy = new HashSet<>(((CtExecutable<?>) toBeInserted.getParent()).getThrownTypes());
                ((CtExecutable<?>) inWhichElement).setThrownTypes(thrownTypesCopy);
```